### PR TITLE
Dirty fix for testing the issue #1401

### PR DESCRIPTION
**Pull request by valib**
The DIDL-Lite sends always the MPEG_PS_PALLocalizedValue but it should be determined if renderer is
capable to use it by using ProtocolInfo.

### CODE REVIEW

**Comment by valib** [comment]
@Nadahar can you help me and give me an advise how to use the renderer ProtocolInfo to determine its capability to play the given media type?

**Comment by Nadahar** [comment]
@valib Each ```Renderer``` (and ```RendererConfiguration```) have a ```deviceProtocolInfo``` field that contains the protocolInfo information for that renderer. It doesn't seem like I made a getter for it, so access is directly via the public field for now. It's easy to add a getter in ```UPnPControl.Renderer``` though, where the field is defined. I was working on an expansion of the protocolInfo stuff that SubJunk didn't have time for me to finish, so I just dropped it then and haven't looked at it since. I might had it planned there, but it's easy to add anyway. 

This isn't a big deal though, as the class held in ```deviceProtocolInfo``` is ```DeviceProtocolInfo```, which is thread-safe. The field is also ```final``` so it's not possible to "set" the field anyway.

That said, your "dirty fix" won't properly fix it. I bet there are renderers that support more than one of these "localized" profiles (I'm pretty sure I've seen that in many log files), we'd have to offer mutiple versions to the renderer (as is, in a way, done today via the renderer setting). The root problem is that when we get the request from the renderer, there's no way to know which of the links were chosen by the renderer (as they all share the same resource id). Using protocolInfo will thus only "solve" it for renderers that only supports one "version".

**Comment by Nadahar** [comment]
@valib When it comes to the use of ```DeviceProtocolInfo``` itself, I don't remember now, but I think it's pretty well documented. It behaves much like a ```List```, but I see now that I didn't implement ```Iterable``` which means you can't use it in a simple ```for``` loop. When I look at it now it also seems that I forgot to include a ```get()``` method, so you'd have to use ```toArray(DeviceProtocolInfoSource<?> type)``` to iterate over the ```ProtocolInfo``` instances.

In the loop, you'd have to check for the profile you're interested in. Those listed are those reported by the renderer. Not all renderers report anything, so the list might be blank.

When you have a ```ProtocolInfo``` instance, getting the profile is easy: ```getDLNAProfileName()```. What you get then is a ```DLNAOrgProfileName``` instance, on which you can call ```getValue()``` to get the string content (everything after ```DLNA.ORG_PN=```).

**Comment by valib** [comment]
@Nadahar agree but it means to add new DLNAResource for each combination of the DlnaOrgPnFlags to be able to be choosen the propper one by renderer. This could be wasting the resources but will be better than current implementaion.

**Comment by Nadahar** [comment]
@valib No, that's not necessary. I've done a kind of a "hack" for this for images, which could be used for other things as well. I add the PN to the link for images, and then I parse it from the link when the request is received. That way I know what profile the renderer picked ;)

**Comment by valib** [comment]
@Nadahar nice but can you add it also for video files?

**Comment by Nadahar** [comment]
@valib It should be possible, take a look at the code used in DLNAResource for ```getURL()``` (AFAICR).

**Comment by valib** [comment]
@UniversalMediaServer/developers can somebody test it?

**Comment by Nadahar** [comment]
@valib I don't think I have any renderer that can test this. You'd need a renderer that use ```ContentFeatures``` and that requests non-US DLNA profiles. The easiest way to test it is probably to get users that has such renderers to do it.

I have a hard time reviewing it as well, as I don't quite understand why you do the changes that you do. You're mixing in changes that seems unrelated to this, and I don't understand what you use ```ProtocolInfo``` for in here.

**Comment by valib** [comment]
@Nadahar the `ProtocolInfo` was in the first attempt when I tried to use it to recognise renderers capability but is not used anymore. I take your "hack" and simply add the PN to the link for video and audio. and check it when `ContentFeatures` are requested.

**Comment by Nadahar** [comment]
@valib Why is that code still part of this PR then?

File: src/main/java/net/pms/dlna/DLNAResource.java [hunk 1]
```diff
@@ -1928,10 +1936,11 @@ private String getDlnaOrgOpFlags(RendererConfiguration mediaRenderer) {
 	 * @return String representation of the DLNA.ORG_PN flags
 	 */
 	@SuppressWarnings("deprecation")
-	private String getDlnaOrgPnFlags(RendererConfiguration mediaRenderer, int localizationValue) {
+	private String getDlnaOrgPnFlags(RendererConfiguration mediaRenderer, int localization) {
```
Comment (SubJunk) [inline src/main/java/net/pms/dlna/DLNAResource.java]: Why rename this parameter here and then reassign it to the old name on line 1943?

**Comment by Nadahar** [comment]
@valib My "problems" with this PR are:

* I don't see the connection between this PR and updating the dependencies. I'm not against updating dependencies, but as always I think there should be a reasonable effort done to check what has changed and estimate the probability that it will affect UMS. I think these changes should at least be in their own commit, and I'd like some words on what considerations have been done about potential issues as a consequence of updating these dependencies.
* I think it's unnecessary messy to leave the code you ended up not using in the PR. As I prefer to review on a commit basis, it is confusing.
* I don't quite understand why you didn't use the existing logic used for images or atleast a very similar logic. We want to make sure these codes are "compatible" and doesn't interfere with each other, so the "formatting rules" should be very strict.

**Comment by SubJunk** [comment]
@valib can this be closed?

**Pull request closed without merging**